src/Plugins/ComplexCore/test/GenerateColorTableTest.cpp

```diff
@@ -0,0 +1,232 @@
+const fs::path k_GrayscalePresetPath = k_TestFilesDir / "Grayscale.txt";
+const fs::path k_HazePresetPath = k_TestFilesDir / "Haze.txt";
+const fs::path k_HSVPresetPath = k_TestFilesDir / "Hsv.txt";
+const fs::path k_JetPresetPath = k_TestFilesDir / "Jet.txt";
+const fs::path k_RainbowBlendedBlackPresetPath = k_TestFilesDir / "RainbowBlendedBlack.txt";
+const fs::path k_RainbowBlendedGreyPresetPath = k_TestFilesDir / "RainbowBlendedGrey.txt";
+const fs::path k_RainbowBlendedWhitePresetPath = k_TestFilesDir / "RainbowBlendedWhite.txt";
+const fs::path k_RainbowDesaturatedPresetPath = k_TestFilesDir / "RainbowDesaturated.txt";
+const fs::path k_RainbowPresetPath = k_TestFilesDir / "Rainbow.txt";
+const fs::path k_XRayPresetPath = k_TestFilesDir / "XRay.txt";
+
+const std::string k_BlackBlueWhitePresetName = "Black, Blue and White";
+const std::string k_BlackOrangeWhitePresetName = "Black, Orange and White";
+const std::string k_BlackBodyRadiationPresetName = "Black-Body Radiation";
+const std::string k_BlueToYellowPresetName = "Blue to Yellow";
+const std::string k_ColdAndHotPresetName = "Cold and Hot";
+const std::string k_GrayscalePresetName = "Grayscale";
+const std::string k_HazePresetName = "Haze";
+const std::string k_HSVPresetName = "hsv";
+const std::string k_JetPresetName = "jet";
+const std::string k_RainbowBlendedBlackPresetName = "Rainbow Blended Black";
+const std::string k_RainbowBlendedGreyPresetName = "Rainbow Blended Grey";
+const std::string k_RainbowBlendedWhitePresetName = "Rainbow Blended White";
+const std::string k_RainbowDesaturatedPresetName = "Rainbow Desaturated";
+const std::string k_RainbowPresetName = "rainbow";
+const std::string k_XRayPresetName = "X Ray";
+
+std::map<std::string, nlohmann::json> ReadPresets()
+{
+  Result<nlohmann::json> result = readRGBPresets(k_PresetsFilePath);
+  COMPLEX_RESULT_REQUIRE_VALID(result);
+
+  std::map<std::string, nlohmann::json> presetsMap;
+  for(const nlohmann::json& preset : result.value())
+  {
+    if(preset.contains("Name") && preset.contains("RGBPoints") && preset["Name"].is_string())
+    {
+      presetsMap.insert({preset["Name"].get<std::string>(), preset});
+    }
+  }
+
+  return presetsMap;
+}
+} // namespace
+
+TEST_CASE("Plugins::GenerateColorTableFilter: Filter Instantiation")
+{
+  const GenerateColorTableFilter filter;
+  Arguments args;
+  DataStructure dataStructure;
+
+  args.insertOrAssign(GenerateColorTableFilter::k_SelectedPreset_Key, std::make_any<nlohmann::json>(nlohmann::json{}));
+  args.insertOrAssign(GenerateColorTableFilter::k_SelectedDataArrayPath_Key, std::make_any<DataPath>(DataPath{}));
+  args.insertOrAssign(GenerateColorTableFilter::k_RgbArrayPath_Key, std::make_any<DataPath>(DataPath{}));
+
+  auto preflightResult = filter.preflight(dataStructure, args);
+  COMPLEX_RESULT_REQUIRE_INVALID(preflightResult.outputActions);
+
+  auto executeResult = filter.execute(dataStructure, args);
+  COMPLEX_RESULT_REQUIRE_INVALID(executeResult.result);
+}
```

Comment:
```suggestion
```

Reply:
@joeykleingers The DynamicFilterInstantiationTest under the complex_test takes care of testing filter instantiation so you should only need this test if you have something specific to check from the results.